### PR TITLE
Add 2.1.1.7 testcase to scenario 2 approach 2.1.1

### PR DESCRIPTION
**Pull request by KasunAratthanage**

### CODE REVIEW

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    private String proxyServiceUrl;
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        super.init();
+        proxyServiceUrl = getProxyServiceURLHttp("2_1_1_7_httpToHttpProtocolTranslationProxy\n");
+        deployCarbonApplication(carFileName);
+    }
+
+    @Test(description = "2.1.1.7 Testcase", enabled = true, dataProvider = "2.1.1.7")
```
Comment (chaminda) [inline product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java]: Please put only the id for the description? word 'Testcase' is redundant. This entry is used for the traceability of test results and testcases in testcase.md

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import org.wso2.carbon.esb.scenario.test.common.HttpConstants;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioConstants;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioTestBase;
+import org.wso2.esb.integration.common.utils.clients.SimpleHttpClient;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+// Test http message to the downstream without altering the content
```
Comment (abeykoon) [inline product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java]: Shall we add the class comment more informatively?

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
+package org.wso2.carbon.esb.scenario.test;
```
Comment (sdkottegoda) [inline product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java]: Missing license header

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java (hunk 2 above)
Comment (sdkottegoda) [inline product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java]: Shall we add doc comment is in the correct format? It should start with /**

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    public void  httpEndpointViaSendMediator(String request, String expectedResponse, String header) throws Exception {
+        log.info("Executing test case 2.1.1.7");
+        SimpleHttpClient httpClient = new SimpleHttpClient();
+        log.info("proxyServiceUrl is set as : " + proxyServiceUrl);
+        Map<String, String> headers = new HashMap<String, String>();
+        headers.put(ScenarioConstants.MESSAGE_ID, header);
+        HttpResponse httpResponse = httpClient.doPost(proxyServiceUrl, headers, request, HttpConstants.MEDIA_TYPE_APPLICATION_XML);
+        String responsePayload = httpClient.getResponsePayload(httpResponse);
+
+        log.info("Actual response received 2.1.1.7: " + responsePayload);
```
Comment (sdkottegoda) [inline product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java]: Let's not log the response

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+        Assert.assertEquals(httpResponse.getStatusLine().getStatusCode(), 200, "HTTP to HTTP transformation failed");
```
Comment (sdkottegoda) [inline product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java]: Don't we need to assert the response?

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/resources/source_files/2.1.1.7/request/2_1_1_7.xml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<?xml version='1.0' encoding='utf-8'?>
```
Comment (sdkottegoda) [inline product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/resources/source_files/2.1.1.7/request/2_1_1_7.xml]: Missing license header

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/resources/source_files/2.1.1.7/response/2_1_1_7.xml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<soap:Envelope
```
Comment (sdkottegoda) [inline product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/resources/source_files/2.1.1.7/response/2_1_1_7.xml]: Missing license header

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+// Test http message to the downstream without altering the content
+public class HttpMessageWithoutAlteringDownstreamTest extends ScenarioTestBase{
+    private final String carFileName = "2_1_1_7_httpToHttpProtocolTranslationCompositeApplication_1.0.0";
+    private String proxyServiceUrl;
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        super.init();
+        proxyServiceUrl = getProxyServiceURLHttp("2_1_1_7_httpToHttpProtocolTranslationProxy\n");
```
[the remark this answers is not in the document]
Comment (KasunAratthanage) [reply product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java]: yes.Fixed it.

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java (hunk 2 above)
Comment (KasunAratthanage) [reply product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java]: Added.

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java (hunk 3 above)
Comment (KasunAratthanage) [reply product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java]: Added.

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java (hunk 1 above)
Comment (KasunAratthanage) [reply product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java]: Fixed.

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java (hunk 4 above)
Comment (KasunAratthanage) [reply product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java]: Fixed.

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/resources/source_files/2.1.1.7/response/2_1_1_7.xml (hunk 7 above)
Comment (KasunAratthanage) [reply product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/resources/source_files/2.1.1.7/response/2_1_1_7.xml]: Added.

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java (hunk 2 above)
Comment (KasunAratthanage) [reply product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java]: Added.

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java (hunk 5 above)
Comment (KasunAratthanage) [reply product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP to HTTP protocol translation/2.1.1-Receive an HTTP request from a client and deliver it over HTTP message to the downstream without altering the content/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java]: Fixed.

**Pull request merged**